core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableSchemaBuilder.java

```diff
@@ -108,6 +122,20 @@ public TableSchema build() {
       schema.setTableProperties(property);
     }
 
+    if (isLocalDictionaryEnabled) {
+      property.put(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE,
+          String.valueOf(isLocalDictionaryEnabled));
+      String localdictionaryThreshold =
+          localDictionaryThreshold.equalsIgnoreCase("0") ? null : localDictionaryThreshold;
+      property.put(CarbonCommonConstants.LOCAL_DICTIONARY_THRESHOLD, localdictionaryThreshold);
+      for (int index = 0; index < allColumns.size(); index++) {
+        ColumnSchema colSchema = allColumns.get(index);
+        if (colSchema.getDataType() == DataTypes.STRING) {
+          colSchema.setLocalDictColumn(true);
```

Comment:
Can you add a comment that why only string columns we enable this ? Why not required for other columns?

Reply:
added comment..Done

Reply:
@babu now we varchar is also supported pls handle the same